src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    operate(action, value) {
+      return action.call(this, value);
+    },
+
+    add(value = 0) {
+      this.result += value;
+
+      return this;
+    },
+
+    subtract(value = 0) {
+      this.result -= value;
+
+      return this;
+    },
+
+    multiply(value = 0) {
+      this.result *= value;
+
+      return this;
+    },
+
+    divide(value = 0) {
+      this.result /= value;
```

Comment:
In the `divide` method, dividing by zero should be handled to prevent potential errors. Consider adding a check for `value === 0` and handling it appropriately, such as returning an error message or setting the result to `Infinity`.